torch/csrc/jit/frontend/sugared_value.h

```diff
@@ -458,15 +458,27 @@ struct MethodValue : public SugaredValue {
     std::vector<const FunctionSchema*> schemas;
     for (const std::string& method_name : method_names_) {
       if (auto class_type = self_->type()->cast<ClassType>()) {
-        Function& method = class_type->getMethod(method_name);
-        try {
-          method.ensure_defined();
-        } catch (const RecursiveMethodCallError&) {
-          throw ErrorReport(loc)
-              << " method '" << method.name() << "' is called recursively. "
-              << "Recursive calls are not supported";
+        Function* method = class_type->findMethod(method_name);
+        if (auto overloaded_method =
+                dynamic_cast<OverloadedFunction*>(method)) {
+          auto matched = overloaded_method->matchOverloadedSchemas(
+              loc, class_type, *f.graph(), argsWithSelf, kwargs);
```

Comment:
I don't remember either but I can give it another try.

Reply:
I think this is because OverloadedFunction has to implement virtual methods defined in Function class. In the case of OverloadedFunction storing other functions inside, it is hard to define the behaviour of methods like run(). We can maybe make OverloadedFunction not subclass from base Function and just keep it as a seperate struct.

Reply:
Ok I added classType as a field inside OverloadedFunction so it knows the other overloaded functions through its' owning class. I think this is similar to how Method object is implemented right now.